Test wxme.rkt in Travis-CI (draft)

gus-massa opened the pull request:
This makes Travis-CI test wxme.rkt, but it has a few problems.

I had to use the minimal installer of Racket because the normal installer installs the default version of `gui-lib` and I wanted to use the version in the current branch. I tried using `raco pkg remove` and `raco pkg install` to replace it, but I get an error. I tried other combinations without success, so I finally decided to install the minimal version and then all the packages I need, but it is slower.

Another problem is that I can't install the `gui-test` package because it links to too many additional packages, so the installation is too slow and I even get a strange error. So ... I'm using the wxme.rkt file from the folder without installing the package. It's not a good idea but it works. In particular this is bad if we want later to test more files in Travis. 

I think that the problem is that `gui-test` uses then [sgl](https://github.com/racket/sgl) package, that builds the docs and then uses a lot of packages. I'm not sure if it's better to split `gui-test` in `gui-test-core` and `gui-test-more` or split the `sgl` package or just keep things as they are now. But I don't understand the details of neither of the packages to be sure.

---

A small detail is that I had to add `(exit 1)` in case wxme.rkt finds an error, because Travis uses the errorlevel to detect failures. With DrDr it's enough to write to the error port, but Travis doesn't care about that. Does this break something? (I also added a `flush-output` because in some test the final line was never printed.)

---

The other commit adds more test for `find-string`. They can be merged independently.

---

I'll push to this branch a  third commit soon. It adds some unnecessary messages to be sure that the test in Travis is using the test in _this_ branch with the code in _this_ branch, instead of the _default_ version of both. The idea is to not merge it, it's only for testing the tests. It also has a bad test, to be sure that Travis detects the error.

gus-massa commented:
This repository doesn't have Travis enabled. In my copy:

[no flush #37](https://travis-ci.org/gus-massa/gui/builds/414848340) -> It's strange. I remember that it didn't show "_22366 tests - all passed_". Perhaps `flush-output` is not necessary, and the process in Travis continue to run an flush all the buffers eventually???

[with flush #38](https://travis-ci.org/gus-massa/gui/builds/414851277) -> Official version

[with error #39](https://travis-ci.org/gus-massa/gui/builds/414859441) -> This is the version that has an error on purpose.

samth commented:
I activated Travis on this repository. You might be interested in how the `typed-racket` repository installs packages to deal with the minimal-racket issue.

gus-massa commented:
I added a small commit to avoid some unnecessary output. It can be squashed with the new set of tests.

@samth I tried copying the travis.yml from `typed-racket` and `rackunit`,  with the obvious modifications of each one, and mixing them and other combinations, but I was not successful. I can retry later. (¿I guess I never used the line with `--skip-installed` [typed-racket/.travis.yml#L18](https://github.com/racket/typed-racket/blob/master/.travis.yml#L18)?)

samth commented:
That line is the one that adds `gui-test`, so it'll be important for you as well.

gus-massa commented:
I tried to copy the `install:` part of the `.travis.yml` file from `typed-racked` as close as possible. It's in another branch https://github.com/gus-massa/gui/commit/9ec329203659d99fbeb7afe67ac7b1534cc671b1 . But I get an [error](https://travis-ci.org/gus-massa/gui/jobs/414953839#L617) 

    collection-path: collection not found
      collection: "gui"
      in collection directories:
      [...]

samth commented:
The problem is that there's no collection named `gui`. It happens that the `typed-racket` package also has a `typed-racket` collection, so it works there.

Probably you went that line to be `raco setup --pkgs gui gui-lib gui-test tex-table`.

gus-massa commented:
@samth  Thanks. I think I fixed it.

gus-massa commented:
I changed the test to run with the the `-l` option, so it use the file from path of library instead .

I also read https://github.com/racket/gui/issues/82 and I added the checks that are in test.rktl. But I usually prefer to call each file individually. I'm using `-eQ` here, but I can use `--drdr` instead. Another option is to drop this part of the PR and add the other tests later.

gus-massa commented:
I think this can be merged now to be included in 7.1. I'm not sure if it's better to include the test in test.rkt or not. Any advice?

mflatt commented:
Loading the test from "test.rkt" seems like a good idea.

I don't have an objection to merging before the v7.1 branch, but unless it seems likely to uncover bugs during the release, I'd be conservative and just wait.

gus-massa commented:
I'll wait then. I don't expect to find something interesting, and I can try to configure my repository to run the test automatically just in case.

> Loading the test from "test.rkt" seems like a good idea.

I'm not sure if this mean that the current version is ok, or it means that it would be better to load each one of the four test that are loaded inside [test.rkt](https://github.com/racket/gui/blob/master/gui-test/tests/gracket/test.rkt) independently.

gus-massa commented:
I think it's better to call each of the test individually instead of using [gracket/test.rkt](https://github.com/racket/gui/blob/master/gui-test/tests/gracket/test.rkt) but they need some tweaking (for example some of them don't have `(require racket/gui)`), so I keep the simple version for now.
